src/internal_telemetry/allocations/allocator/token.rs

```diff
@@ -20,20 +19,21 @@ thread_local! {
 
 /// The identifier that uniquely identifiers an allocation group.
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
-pub struct AllocationGroupId(NonZeroUsize);
+pub struct AllocationGroupId(u8);
```

Comment:
This should be a `NonZero*` variant of whatever underlying integer type we use. Using `NonZero*` is what gives us the correct invariants to depend on never having a valid group ID as zero.

Granted it doesn't matter right now since allocation tracing can only be changed at process start, but it's an entirely zero-cost mechanism, so we should ay the groundwork more carefully for that eventual future.